### PR TITLE
Adds a configurable tracing output instead of hardcoding stderr

### PR DESCRIPTION
**Pull request by jneem**
I'm not super sure about the API; it's a little painful to have to thread the tracing writer everywhere. Also, what's the story/plan with the crate's API stability?

I've checked that the wasm tracing works on the website, if you init the repl with `repl_init(console.log)`

### CODE REVIEW

**Comment by jneem** [comment]
Sorry, I forgot to link the original [issue](https://github.com/tweag/nickel/issues/1069)

**Review by vkleen — COMMENTED** [review]
Apart from my comment in `flake.nix` this functionality looks good to me. I'm not hitting approve just because we may want to discuss the `Program` API change a bit more before merging.

**Review by yannham — APPROVED** [review]
Besides a few comments, looks good to me.

> I'm not super sure about the API; it's a little painful to have to thread the tracing writer everywhere.

It's not ideal, but it's not too bad I think. We need to trace it through `Program` and `VirtualMachine`, but from there it's accessible via the virtual machine, which encompasses most of evaluation functions. Ideally we would like a `Reader` monad or implicit parameters but I find that putting stuff inside `VirtualMachine` isn't too far from that in practice. We might introduce a form of `ProgramConfing` or `VirtualMachineConfig` at some point, with builders, to make the creation slightly smoother.

> Also, what's the story/plan with the crate's API stability?

For now, there is none :stuck_out_tongue: we just released 1.0 so we have to be careful of correctly versioning nickel-the-library-and-not-the-binary, but for now we focus on backward compatibility and stability of the language, not the library. We haven't thought this through for the crate's API so that would be risky to commit to whatever currently exists, which might be in part accidental. I guess this will come in later versions of nickel-the-library-crate.

File: utilities/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
 pub mod annotated_test;
 pub mod bench;
 pub mod test_program;
+
+pub use nickel_lang;
```
[the remark this answers is not in the document]
Comment (jneem) [reply utilities/src/lib.rs:5]: Yes, for some reason I didn't dig into, in the deserialization tests `super::RustDeserializationError`  didn't equal the deserialization error type coming from the `nickel_lang_utilities` crate, even though they seem like two different paths to the same thing. So to get the tests to work I just made everything accessible through `nickel_lang_utilities`

[the remark this answers is not in the document]
Comment (yannham) [reply utilities/src/lib.rs:5]: Ah, I think we had this issue before. I vaguely remember it was due to `RustDeserializationError` being defined in two different parts of `serde`, and both were imported, or something like that. I suspect using "absolute" paths for the type might fix that, but I'm not 100% sure.

[the remark this answers is not in the document]
Comment (yannham) [reply utilities/src/lib.rs:5]: Anyway, whether this works or not, I let you do what you think is best and merge

**Pull request merged**